src/SpatialNavigation.ts

```diff
@@ -1189,18 +1212,21 @@ class SpatialNavigationService {
         return;
       }
 
-      forEach(this.focusableComponents, (component) => {
-        if (component.parentFocusKey === focusKey && component.focusable) {
-          // eslint-disable-next-line no-param-reassign
-          component.parentFocusKey = parentFocusKey;
-        }
-      });
```

Comment:
This logic was a workaround to the root issue here. Theoretically if the parent is removed, there is no need to update its children parent keys, they will get removed as well.